src/sentry/receivers/owners.py

```diff
@@ -0,0 +1,80 @@
+from django.db.models.signals import pre_delete, pre_save
+
+from sentry.models.organization import Organization
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.team import Team
+from sentry.roles import organization_roles
+
+
+def _assert_org_has_owner_not_from_team(organization, top_role):
+    assert organization.member_set.filter(
+        role=top_role
+    ).exists(), "An organization must have at least one owner"
+
+
+def prevent_demoting_last_owner(instance: OrganizationMember, **kwargs):
+    # if a member is being created
+    if instance.id is None:
+        return
+
+    try:
+        member = OrganizationMember.objects.get(id=instance.id)
+    except OrganizationMember.DoesNotExist:
+        return
+
+    # member is the last owner and the update will remove the last owner
+    assert not (
+        member.is_only_owner()
+        and organization_roles.get_top_dog().id not in instance.get_all_org_roles()
```

Comment:
What does this check for?
```organization_roles.get_top_dog().id not in instance.get_all_org_roles()```

Reply:
oh actually i don't think i need that check. it's already being done in `is_only_owner`

Reply:
scratch that. `member` refers to the original object before it has been modified, and `instance` refers to the object that has been modified. so i'm checking that when the role is changed to what it is in `instance` that we still have an owner

we want to prevent the case where the `member` is the only owner and the member's role is being changed to something other than owner